File: commands/misc/userinfo.js
const moment = require('moment');

module.exports = {
    name: 'userinfo',
    description: 'This allows users to find out more information about themselves or another user they ping or provide the ID for.',
    aliases: ['user', 'person', 'useri', 'user-info'],
    usage: '++userinfo',
    inHelp:'yes',
    example: '++userinfo',
    execute (message) {

      if (message.reference === null) { // just a regular message
        const member = message.mentions.users.first() || message.author || message.member;
        let userinfo = {};
        userinfo.createdat = member.createdAt;
        userinfo.discrim = member.discriminator;
        userinfo.id = member.id;
        userinfo.tag = member.tag;
        userinfo.nick = message.guild.members.cache.get(member.id).displayName;
        userinfo.uname = member.username;
        userinfo.avatar = member.displayAvatarURL({ dynamic: true });
        const memberMention = message.mentions.members.first() || message.member;
        const rolesOfTheMember = memberMention.roles.cache.filter(r => r.name !== '@everyone').map(role => role.name).join(', ');
        console.log(userinfo)

        var myInfo = {
          color: '#7DCE87',
          author: {
            name: userinfo.uname,
            icon_url: userinfo.avatar
          },
          title: 'About this user...',
          thumbnail: {
            url: userinfo.avatar
          },
          fields: [
            {
              name: 'Username',
              value: userinfo.uname,
              inline: true,
            },
            {
              name: 'Discriminator',
              value: userinfo.discrim,
              inline: true,
            },
            {
              name: 'Nickname',
              value: userinfo.nick,
              inline: true,
            },
            {
              name: 'User ID',
              value: userinfo.id,
              inline: true,
            },
            {
              name: 'Tag',
              value: userinfo.tag,
              inline: true,
            },
            {
              name: 'Joined server on',
              value: `\`${moment(message.guild.members.cache.get(member.id).joinedAt).format('MMM DD YYYY')}\``,
              inline: true,
            }, 
            {
              name: 'Joined Discord on',
              value: `\`${moment(member.createdAt).format('MMM DD YYYY')}\``,
              inline: true,
            },
            {
              name: 'Roles',
              value: rolesOfTheMember,
              inline: true,
            }
          ],
          timestamp: new Date(),
          footer: {
            text: 'If anything is wrong, please report this!',
            icon_url: message.guild.iconURL({ dynamic: true })
          }
        }
        message.channel.send({ embeds: [myInfo] });
        //console.log(myInfo)

        } else {
          const member = message.mentions.repliedUser;
          let userinfo = {};
          userinfo.createdat = member.createdAt;
          userinfo.discrim = member.discriminator;
          userinfo.id = member.id;
          userinfo.tag = member.tag;
          userinfo.nick = message.guild.members.cache.get(member.id).displayName;
          userinfo.uname = member.username;
          userinfo.avatar = member.displayAvatarURL({ dynamic: true });
          const memberMention = message.mentions.members.first() || message.member;
          const rolesOfTheMember = memberMention.roles.cache.filter(r => r.name !== '@everyone').map(role => role.name).join(', ')
  
          var myInfo = {
            color: '#7DCE87',
            author: {
              name: userinfo.uname,
              icon_url: userinfo.avatar
            },
            title: 'About this user...',
            thumbnail: {
              url: userinfo.avatar
            },
            fields: [
              {
                name: 'Username',
                value: userinfo.uname,
                inline: true,
              },
              {
                name: 'Discriminator',
                value: userinfo.discrim,
                inline: true,
              },
              {
                name: 'Nickname',
                value: userinfo.nick,
                inline: true,
              },
              {
                name: 'User ID',
                value: userinfo.id,
                inline: true,
              },
              {
                name: 'Tag',
                value: userinfo.tag,
                inline: true,
              },
              {
                name: 'Joined server on',
                value: `\`${moment(message.guild.members.cache.get(member.id).joinedAt).format('MMM DD YYYY')}\``,
                inline: true,
              }, 
              {
                name: 'Joined Discord on',
                value: `\`${moment(member.createdAt).format('MMM DD YYYY')}\``,
                inline: true,
              },
              {
                name: 'Roles',
                value: rolesOfTheMember,
                inline: true,
              }
            ],
            timestamp: new Date(),
            footer: {
              text: 'If anything is wrong, please report this!',
              icon_url: message.guild.iconURL({ dynamic: true })
            }
          }

          message.channel.send({ embeds: [myInfo] });
        }
        
        
    }

};

File: lib/src/commands/misc/userinfo.dart
import 'package:intl/intl.dart';
import 'package:nyxx/nyxx.dart';
import 'package:nyxx_commands/nyxx_commands.dart';

final userInfo = ChatCommand(
  'user-info',
  'Get information about a user',
  aliases: ['user', 'person', 'useri', 'userinfo'],
  id('user-info', (
    IChatContext context, [
    @Description('The user to fetch information for') IUser? user,
  ]) async {
    if (context is MessageChatContext) {
      // Command was triggered with a reply, get the info for the user that was replied to.
      IMessageAuthor? author = context.message.referencedMessage?.message?.author;

      if (author is IUser) {
        user ??= author;
      }
    }

    // Default to the user that triggered the command.
    user ??= context.user;
    IMember? member = await context.guild?.fetchMember(user.id);

    final joinedDiscord = DateFormat('d/M/y').format(user.id.timestamp);
    final joinedServer = DateFormat('d/M/y').format(member?.joinedAt ?? user.id.timestamp);

    final roles =
        member?.roles.map((role) => role.getFromCache()?.name).whereType<String>().join(', ');

    final embed = EmbedBuilder()
      ..color = DiscordColor.green
      ..addAuthor((author) {
        author.name = user!.username;
        author.iconUrl = user.avatarURL();
      })
      ..title = 'About this user...'
      ..thumbnailUrl = user.avatarURL()
      ..addField(name: 'Username', content: user.username, inline: true)
      ..addField(name: 'Discriminator', content: user.discriminator, inline: true)
      ..addField(
        name: 'Nickname',
        content: member?.nickname ?? user.username,
        inline: true,
      )
      ..addField(name: 'User ID', content: user.id.toString(), inline: true)
      ..addField(name: 'Tag', content: user.tag, inline: true)
      ..addField(name: 'Joined server on', content: joinedServer, inline: true)
      ..addField(name: 'Joined Discord on', content: joinedDiscord, inline: true)
      ..timestamp = DateTime.now()
      ..addFooter((footer) {
        footer.text = 'If anything is wrong, please report this!';
        footer.iconUrl = context.guild?.iconURL();
      });

    if (roles != null && roles.isNotEmpty) {
      embed.addField(name: 'Roles', content: roles, inline: true);
    }

    await context.respond(MessageBuilder.embed(embed));
  }),
);
